src/Parsing/Impl/Ast/PythonAst.cs

```diff
@@ -56,12 +58,12 @@ public sealed class PythonAst : ScopeStatement, ILocationConverter {
                 offset += a.NewLineLocations.Length + 1;
             }
             CommentLocations = comments.ToArray();
+            ScopeInfo = new AstScopeInfo(this);
         }
 
         public Uri Module { get; }
         public NewLineLocation[] NewLineLocations { get; private set; }
         public SourceLocation[] CommentLocations { get; private set; }
-        public override string Name => "<module>";
```

Comment:
I've seen a lot of removal of `Name` in favor of `ScopeName`. Are you certain this would be a backwards compatible change?

Reply:
Same here - is there a reason? Parser is public and used outside of the language server - IntelliCode and Python Tools for VS. We can't just change members.

Reply:
Given this is an override being removed, I'm not sure it's "gone" per se, but I haven't yet found the code that would redirect `ScopeName` to `Name`. My impression is that `INode` should have `string Name` and `ScopeName` can be dropped entirely.

Reply:
Ok I like the idea of adding Name to INode

Reply:
If ScopeName gets eliminated, and everything continues to use Name, will there be any reason to have INode at all?

Reply:
Well, given the number of things that IScopeNode picks up, maybe it would...

Reply:
I have INode in order to have IScopeNode implement it and handle interfaces that used to take in Node and were given ScopeStatements so that they now take in INode and are given IScopeNodes

Reply:
Might actually have to leave Name in IScopeNode. Otherwise for other nodes, I could redirect Name to NodeName.

Reply:
Leaving stuff in IScopeNode is probably fine, we're just trying to eliminate changes that look backward compatible, or interfaces that could be avoided.

Reply:
On the other hand it becomes confusing. `NodeName` != `Name`, consider `NameExpression.Name` vs `NameExpression.NodeName`

Reply:
Then yeah I left `Name` field in `IScopeNode` so it only has name if it actually has something different than node name